File: apps/api/src/application/imports/get-import-job.ts
import { NotFoundError } from '@/domain/shared/domain-error'

import type { ImportJobRepository } from '@/application/ports/import-job-repository'
import type { ImportJob } from '@/domain/import/import-job'

export class GetImportJob {
  constructor(private readonly imports: ImportJobRepository) {}

  async execute(id: string): Promise<ImportJob> {
    const job = await this.imports.findById(id)
    if (!job) {
      throw new NotFoundError('import', id)
    }
    return job
  }
}

File: apps/api/src/ecommerce_api/application/imports/get_import_job.py
from ecommerce_api.application.ports.import_job_repository import ImportJobRepository
from ecommerce_api.domain.errors import NotFoundError
from ecommerce_api.domain.import_job import ImportJob


class GetImportJob:
    def __init__(self, imports: ImportJobRepository) -> None:
        self._imports = imports

    async def execute(self, id: str) -> ImportJob:
        job = await self._imports.find_by_id(id)
        if job is None:
            raise NotFoundError('import', id)
        return job
